airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -113,6 +115,10 @@ def retry_factor(self) -> float:
     def authenticator(self) -> HttpAuthenticator:
         return self._authenticator
 
+    @property
+    def availability_strategy(self) -> Optional[AvailabilityStrategy]:
```

Comment:
nit: Would we want a test that asserts the default  `availability_strategy` of an HttpStream is an `HttpAvailabilityStrategy` instance.

Reply:
That is handled [here](https://github.com/airbytehq/airbyte/pull/21924/files#diff-0be73143af526e026bfe75f6fc8e166b0431f7fc2ccb847d3410560246d3f24cR61)

```
http_stream = MockHttpStream()
    assert isinstance(http_stream.availability_strategy, HttpAvailabilityStrategy)
```

Where `MockHttpStream()` subclasses `HttpStream` and before this PR, was given the default explicitly:

```
        # TODO (Ella): Remove explicit definition when turning on default
        @property
        def availability_strategy(self) -> Optional["AvailabilityStrategy"]:
            return HttpAvailabilityStrategy()
```
but the explicit definition [has been removed in this PR](https://github.com/airbytehq/airbyte/pull/21924/files#diff-0be73143af526e026bfe75f6fc8e166b0431f7fc2ccb847d3410560246d3f24cL43-L47), so the test now checks that it inherits that default